Doc/library/sqlite3.rst

```diff
@@ -2614,6 +2590,49 @@ With some adjustments, the above recipe can be adapted to use a
 instead of a :class:`~collections.namedtuple`.
 
 
+.. _sqlite3-howto-encoding:
+
+How to handle non-UTF-8 text encodings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` uses :class:`str` to adapt SQLite values
+with the ``TEXT`` data type.
+This works well for UTF-8 encoded text, but it might fail for other encodings
+and invalid UTF-8.
+You can use a custom :attr:`~Connection.text_factory` to handle such cases.
+
+Because of SQLite's `flexible typing`_, it is not uncommon to encounter table
+columns with the ``TEXT`` data type containing non-UTF-8 encodings,
+or even arbitrary data.
+To demonstrate, let's assume we have a database with ISO-8859-2 (Latin-2)
+encoded text, for example a table of Czech-English dictionary entries.
+Assuming we now have a :class:`Connection` instance :py:data:`!con`
+connected to this database,
+we can decode the Latin-2 encoded text using this :attr:`~Connection.text_factory`:
+
+.. testcode::
+
+   con.text_factory = lambda data: str(data, encoding="latin2")
+
+For invalid UTF-8 or arbitrary data in stored in ``TEXT`` table columns,
+you can use the following technique, borrowed from the :ref:`unicode-howto`:
+
+.. testcode::
+
+   con.text_factory = lambda data: str(data, errors="surrogateescape")
+
+.. note::
+
+   Strings containing surrogate escapes must be treated with care.
+   For example, you cannot pass them back to SQLite,
+   since the :mod:`!sqlite3` module API does not support strings
+   containing surrogate escape codes.
```

Comment:
```suggestion
   The :mod:`!sqlite3` module API does not support strings
   containing surrogates.
```
I would simply say this then.  This will make the users aware of this limitation without going in too much details, so that they will remove the surrogates and try again.  If there are situations where surrogates are allowed (e.g. can I have them in a db and read them?) then it might be worth to specify that.

Reply:
The example just above this text shows how to deal with strings w/surrogates in the DB. I think your proposal works swell. Keep it simple 🪄

Reply:
Alternatively, one could remove this warning and mention this limitation explicitly for each API.